fe/fe-core/src/main/java/com/starrocks/http/rest/TransactionLoadAction.java

```diff
@@ -114,20 +122,141 @@ public class TransactionLoadAction extends RestBaseAction {
     // Map operation name to metrics
     private final Map<TransactionOperation, OpMetrics> opMetricsMap = new HashMap<>();
 
-    private final ReadWriteLock txnNodeMapAccessLock = new ReentrantReadWriteLock();
-    private final Map<String, Long> txnNodeMap = new LinkedHashMap<>(512, 0.75f, true) {
-        @Override
-        protected boolean removeEldestEntry(Map.Entry<String, Long> eldest) {
-            return size() > (GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalBackendNumber() +
-                    GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getTotalComputeNodeNumber()) * 512;
-        }
-    };
+    private final TransactionLoadLabelCache txnNodeMap = new TransactionLoadLabelCache();
 
     public TransactionLoadAction(ActionController controller) {
         super(controller);
         initMetrics();
     }
 
+    @VisibleForTesting
+    public TransactionLoadLabelCache getTxnNodeMap() {
+        return txnNodeMap;
+    }
+
+    public static class TransactionLoadLabelCache {
```

Comment:
could put this class in a separate java file, and add test for it which do not dependent on transaction stream load. Maybe this cache can be reused by other scenarios in the future.